lucetc/src/decls.rs

```diff
@@ -418,10 +407,11 @@ impl<'a> ModuleDecls<'a> {
     }
 
     pub fn get_func(&self, func_index: UniqueFuncIndex) -> Result<FunctionDecl<'_>, Error> {
```

Comment:
The only Error this method throws is FunctionIndexError, which should only happen if the caller provides a bogus func_index. I'd rather it just return an Option<FunctionDecl<'_>> and make it the caller's problem. Then you can delete the FunctionIndexError variant

Reply:
I took your suggestion to have `get_func` return an Option.  Unfortunately, the way I interpreted your suggestion only moves the use of `FunctionIndexError`.  Let me know how I might improve this so I can follow your suggestion 100%.